src/hotspot/share/jvmci/jvmciCompilerToVM.cpp

```diff
@@ -1058,9 +1058,9 @@ C2V_END
 C2V_VMENTRY_0(jlong, getMaxCallTargetOffset, (JNIEnv* env, jobject, jlong addr))
   address target_addr = (address) addr;
   if (target_addr != 0x0) {
-    int64_t off_low = (int64_t)target_addr - ((int64_t)CodeCache::low_bound() + sizeof(int));
-    int64_t off_high = (int64_t)target_addr - ((int64_t)CodeCache::high_bound() + sizeof(int));
-    return MAX2(ABS(off_low), ABS(off_high));
```

Comment:
Wouldn't this trigger a warning with -Wsign-conversion enabled?

Reply:
Maybe? I added `checked_cast` in this place in new commit.

Reply:
checked_cast isn't really helping here.  It doesn't work when the input is the same size as the output.  We need a better way to check for bad input or bad outputs.  I suggested before doing those checks in the abs/ABS implementation, but we could also have something like u2i() (or u2s?) for unsigned to signed conversions instead of checked_cast.